service/turn.rest/src/main/java/org/kaazing/gateway/service/turn/rest/TurnRestCredentialGenerator.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+import javax.security.auth.Subject;
+
+import org.kaazing.gateway.service.ServiceProperties;
+
+public interface TurnRestCredentialGenerator {
+    
+    /**
+     * 
+     * @param options nested property containing a shared secret key, timestamp/username
+     * separator to be used in the generated username, etc.  
+     */
+    public void init(ServiceProperties options);
+
+    /**
+     * 
+     * @param parameterUsername username passed in as HTTP parameter if present, null otherwise 
+     * @param subject Subject representing the currently logged in user
+     * @return TurnRestCredentials object 
+     */
+    public TurnRestCredentials generateCredentials(String parameterUsername, Subject subject);
```

Comment:
Need to think about the username and api key future compatibility on the parameters